### PR TITLE
Prevent Partytown from hijacking history API

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Binds the `pushState` and `replaceState` methods which break Partytown with Firefox.
- Fixes https://github.com/withastro/astro/issues/8862

## Testing

- Manually

## Docs

N/A, bug fix

### CODE REVIEW

**Comment by matthewp** [comment]
!preview vt-partytown

File: packages/astro/src/transitions/router.ts [hunk 1]
```diff
@@ -9,12 +9,17 @@ type State = {
 };
 type Events = 'astro:page-load' | 'astro:after-swap';
 
+// Create bound versions of pushState/replaceState so that Partytown doesn't hijack them,
+// which breaks Firefox.
+const pushState = history.pushState.bind(history);
+const replaceState = history.replaceState.bind(history);
+
```
Comment (martrapp) [inline packages/astro/src/transitions/router.ts:16]: Looks good to me. I have no idea, how party town scripts are injected. But these assignments should be early enough as `router.ts` is loaded through `ViewTransitions.astro` from the `<head>`.

**Review by martrapp — APPROVED** [review]
LGTM

File: packages/astro/src/transitions/router.ts (hunk 1 above)
Comment (martrapp) [reply packages/astro/src/transitions/router.ts:16]: Should be guarded with `if (inBrowser)...`

Comment (matthewp) [reply packages/astro/src/transitions/router.ts:16]: Does the router ever run outside of a browser?

Comment (martrapp) [reply packages/astro/src/transitions/router.ts:16]: not run but loaded in SSR, see #8771

Comment (martrapp) [reply packages/astro/src/transitions/router.ts:16]: .. and in that failing test :)

**Pull request merged**